morphia/src/main/java/org/mongodb/morphia/query/QueryImpl.java

```diff
@@ -45,26 +48,16 @@
     private EntityCache cache;
     private boolean validateName = true;
     private boolean validateType = true;
-    private BasicDBObject projections;
     private Boolean includeFields;
-    private BasicDBObject sort;
-    private BasicDBObject max;
-    private BasicDBObject min;
-    private int offset;
-    private int limit = -1;
-    private int batchSize;
-    private String indexHint;
     private BasicDBObject baseQuery;
-    private boolean snapshotted;
-    private boolean noTimeout;
-    private boolean tail;
-    private boolean tailAwaitData;
-    private ReadPreference readPref;
-    private Integer maxScan;
-    private Long maxTime;
-    private TimeUnit maxTimeUnit;
-    private String comment;
-    private boolean returnKey;
+    private FindOptions options;
+
+    FindOptions getOptions() {
+        if (options == null) {
```

Comment:
My concern is that it's error-prone, as it requires all internal users of options to get it through this method, and it would be easy to forget to do that.  

But it's up to you.  If you don't think it's a problem, then leave as is.